src/shared/utils/billing.js

```diff
@@ -136,6 +146,21 @@ export const findSentryPlans = ({ plans }) => {
   }
 }
 
+// This fn expects to use the availablePlans resolver, not the rest endpoint, hence the plan.planName instead of plan.value
+export const findTeamPlans = ({ availablePlans }) => {
+  const teamPlanMonth = availablePlans?.find(
+    (plan) => plan.planName === Plans.USERS_TEAMM
+  )
+  const teamPlanYear = availablePlans?.find(
+    (plan) => plan.planName === Plans.USERS_TEAMY
+  )
+
+  return {
+    teamPlanMonth,
+    teamPlanYear,
+  }
+}
```

Comment:
I'm also not the biggest fan of having a `find<function>` that works differently from the rest, but there's some limitation with the current use of our `usePlans` hook through REST. Here's the explanation (brace yourself):

Historically, we've had this function that determines the plans available to you, https://github.com/codecov/codecov-api/blob/main/billing/helpers.py#L27 (32-34). Pretty much returns all the plans we commonly use in our client. Before sentry, we never separated what plans were available to an org/user, they all had the same. This fn just returned the "available plans codecov exposes to the users", filtering out obsolete ones.

Sentry requirement came in (36-38) couple of months ago, where the plans changed based on the **owner** that requests this, where the owner is the entity that holds orgs (so adrian-codecov to codecov). Then, we adjusted this function to do so. This is easy because **every** request, REST or GQL, knows what the owner/requesting user is as part of our authentication process, see here https://github.com/codecov/codecov-api/blob/main/codecov_auth/middleware.py#L67. 

That is the present day.

Comes: team tier. Team plans are only exposed to `organizations` that follow certain criteria (see here for the specific logic https://github.com/codecov/codecov-api/blob/main/plan/service.py#L118). We're suddenly introducing this notion of an organization us to know if we can/can't surface a team plan.

What's the problem with REST? The plans endpoint does not know have the context of an org, see the plans url https://github.com/codecov/codecov-api/blob/main/plan/service.py#L118, as opposed to something like https://github.com/codecov/codecov-api/blob/main/api/internal/urls.py#L64 that requires the owner as a parameter. Because of this, as of now, the REST endpoint can't support returning team plans. This makes sense because historically, we never made plans available dependent on the organization nor the owner, and requirements have changed since.

Why does it work on GQL? Because we expose the logic through the PlanService, which takes in an organization as a parameter, and also has access to the owner through the request. 

So, this is the problem 😮‍💨.

Solutions?

1) We port to GQL as the solution is there and we're also favouring moving away from REST endpoints off of Gazebo. I would make a separate PR, prior to this, to change every instance of the REST plans route and repurpose the `usePlans` hook to fetch `owner > plan > availablePlans` via GQL. I'd also adjust all the utils to adhere to this new resolver. This is my preferred approach.
2) We add the owner param to the plans REST route. We'd have to make a mock route, like plans-temp, add the owner param to that one, have the frontend use plans-temp while we fix plans, and then move that back to plans. This is a bit more work imo, and we'll eventually want to do 1 by porting everything to GQL. 

Wanted to hear your thoughts about this, although if I had to choose, I'd be in favor of moving to GQL

Reply:
No worries, I did not know of this limitation at the time of writing, and it's been a hot minute since I've used `usePlans`, I'm definitely in favour of moving off of the REST API as much as we can.